src/plots/geo/constants.js

```diff
@@ -135,12 +135,23 @@ params.fillLayers = ['ocean', 'land', 'lakes'];
 // base layer with a only a line color
 params.lineLayers = ['subunits', 'countries', 'coastlines', 'rivers', 'frame'];
 
-// all base layers - in order
-params.baseLayers = [
+params.layers = [
+    'bg',
     'ocean', 'land', 'lakes',
     'subunits', 'countries', 'coastlines', 'rivers',
-    'lataxis', 'lonaxis',
-    'frame'
+    'lataxis', 'lonaxis', 'frame',
+    'backplot',
+    'frontplot'
+];
+
+params.layersForChoropleth = [
+    'bg',
+    'ocean', 'land',
+    'subunits', 'countries', 'coastlines',
+    'lataxis', 'lonaxis', 'frame',
+    'backplot',
+    'rivers', 'lakes',
+    'frontplot'
```

Comment:
I'm a bit confused by the choropleth distinction here... it seems to me that either:
- adding a choropleth (even if it only shows up in some other part of the plot) will reorder other layers such that there are visual changes, which would be bad, right?
- if there are no visual changes, we should be able to use the same layer structure whether or not there is a choropleth present - with at most an extra layer just for choropleth, separate from backplot (though I doubt it, seems like choropleth is the only thing that uses backplot currently, right?)

Reply:
Yeah, pasting what I wrote above under _open items_:

> Base layer ordering changes whenever a choropleth trace is present on the subplot (see this [block](https://github.com/plotly/plotly.js/commit/4c1ddb80dd39ae6d1ba5d51d21b95b36919a8bc5#diff-b4732bdd35eeb4830531de6212e422e2R239)). This logic is meant to preserve backward compatibility (i.e. to mimic what that [block](https://github.com/plotly/plotly.js/blob/60b445d2916547f1d1a548d64209989688d7a617/src/traces/choropleth/plot.js#L65-L72) does in the current `master`). But really, we should make the base layer ordering consistent for all trace type combinations in v2. But in the meantime, we should make the base layer ordering customizable (similar to `axis.layer`).

Reply:
Ah OK, missed that note, thanks. 👍